base/poll.jl

```diff
@@ -56,293 +26,381 @@ const FD_TIMEDOUT = 4
 
 isreadable(f::FDEvent) = f.readable
 iswritable(f::FDEvent) = f.writable
-FDEvent() = FDEvent(false,false,false)
+FDEvent() = FDEvent(false, false, false)
 FDEvent(flags::Integer) = FDEvent((flags & UV_READABLE) != 0,
                                   (flags & UV_WRITABLE) != 0,
                                   (flags & FD_TIMEDOUT) != 0)
-fdtimeout() = FDEvent(false,false,true)
+fdtimeout() = FDEvent(false, false, true)
 
-#Wrapper for an OS file descriptor (for Windows)
-@windows_only immutable WindowsRawSocket
-    handle::Ptr{Void}   # On Windows file descriptors are HANDLE's and 64-bit on 64-bit Windows...
-end
 
-@windows_only export WindowsRawSocket
+type FileMonitor
+    handle::Ptr{Void}
+    file::ByteString
+    notify::Condition
+    active::Bool
+    function FileMonitor(file::AbstractString)
+        handle = Libc.malloc(_sizeof_uv_fs_event)
+        this = new(handle, file, Condition(), false)
+        associate_julia_struct(handle, this)
+        err = ccall(:uv_fs_event_init, Cint, (Ptr{Void}, Ptr{Void}), eventloop(), handle)
+        if err < 0
+            Libc.free(handle)
+            throw(UVError("PollingFileWatcher", err))
+        end
+        finalizer(this, uvfinalize)
+        this
+    end
+end
 
 abstract UVPollingWatcher
 
 type PollingFileWatcher <: UVPollingWatcher
     handle::Ptr{Void}
-    file::AbstractString
-    open::Bool
+    file::ByteString
+    interval::UInt32
     notify::Condition
-    cb::Callback
-    function PollingFileWatcher(cb, file)
+    active::Bool
+    function PollingFileWatcher(file::AbstractString, interval::Float64=5.007) # same default as nodejs
+        if interval < 2
+            warn("For maximum portability, use multi-second intervals with PollingFileWatcher")
```

Comment:
I just prefer not to annoy people with warnings. Realistically, somebody is not going to happily run a program that scrolls that warning repeatedly, so it might as well be an error. Especially since the warning depends directly on a passed argument, we need to decide whether the argument value is allowed or not.